src/core/reactor_backend.cc

```diff
@@ -1486,10 +1492,98 @@ class reactor_backend_uring final : public reactor_backend {
         delete pfd;
     }
     virtual future<std::tuple<pollable_fd, socket_address>> accept(pollable_fd_state& listenfd) override {
-        return _r.do_accept(listenfd);
+        if (is_speculated(listenfd, POLLIN | POLLOUT)) {
+            listenfd.maybe_no_more_recv();
+            socket_address sa;
+            auto maybe_fd = listenfd.fd.try_accept(sa, SOCK_CLOEXEC);
+            if (maybe_fd) {
+                listenfd.speculate_epoll(EPOLLIN);
+                pollable_fd pfd(std::move(*maybe_fd), pollable_fd::speculation(EPOLLOUT));
+                return make_ready_future<std::tuple<pollable_fd, socket_address>>(std::move(pfd), std::move(sa));
+            }
+        }
+        class accept_completion final : public io_completion {
+            pollable_fd_state& _listenfd;
+            socket_address _sa;
+            promise<std::tuple<pollable_fd, socket_address>> _result;
+        public:
+            accept_completion(pollable_fd_state& listenfd)
+                : _listenfd(listenfd) {}
+            void complete(size_t fd) noexcept final {
+                _listenfd.speculate_epoll(EPOLLIN);
+                pollable_fd pfd(file_desc::from_fd(fd), pollable_fd::speculation(EPOLLOUT));
+                _result.set_value(std::move(pfd), std::move(_sa));
```

Comment:
Do we have tests that server_socket::abort_accept() causes server_socket::accept() to return? If not please add a test.

The code looks like it should work, but better to be sure.

Reply:
yeah. actually, i added this just to fix the unexpected invalid_argument (errno 22) exception thrown when i stopped the httpd server.

now since the accept+connect patch will be dropped in this PR. i will continue working on it in another one.